File: scala/06-msg-via-router/src/main/scala/org/elu/scala/akka/Worker.scala
package org.elu.scala.akka

import akka.actor.Actor

class Worker extends Actor {
  import Worker._

  override def receive: Receive = {
    case msg: Work =>
      println(s"I received Work message and my ActorRef: $self")
  }
}

object Worker {
  case class Work()
}

File: kotlin/06-msg-via-router/src/main/kotlin/org/elu/kotlin/akka/Worker.kt
package org.elu.kotlin.akka

import akka.actor.AbstractActor
import akka.japi.pf.ReceiveBuilder

class Work

class Worker : AbstractActor() {
  override fun createReceive(): Receive =
      ReceiveBuilder()
          .match(Work::class.java, this::onMessage)
          .build()

  private fun onMessage(msg: Work) {
    println("I received Work message and my ActorRef: $self")
  }
}
